OpenRA.Mods.Common/SpriteLoaders/PngSheetLoader.cs

```diff
@@ -31,6 +30,26 @@ public PngSheetMetadata(Dictionary<string, string> metadata)
 		}
 	}
 
+	public class EmbeddedSpritePalette
```

Comment:
As D2k is still *the* main user of this it makes no sense for it to be in `PngSheetLoader`. Can we give it its own file please?

Reply:
It's part of the png(sheet) format so IMO it does make sense here, but putting it in its own file is probably fine too.

Reply:
`PngSheetLoader` and `R8Loader` both use it identically in their `TryParseSprite` methods and it (and `PaletteFromEmbeddedSpritePalette`) is (now) generic enough that it doesn't need to be tied to a specific format.
Also class-per-file is a very basic convention that is just a good idea.
Would be nice to give it its own file.